docs/labs/flockercontainers.rst

```diff
@@ -0,0 +1,97 @@
+.. _flocker-containers:
+
+=============================
+Running Flocker in Containers
+=============================
+
+For some environments, such as CoreOS, and ECS you must install Flocker inside a container.
+
+.. warning::
+   Installing Flocker in containers is useful for environments such as CoreOS, or Amazon ECS, but please note that doing so is currently experimental.
+
+You can find the relevant Docker Hub images here:
+
+* `Dataset Agent <https://hub.docker.com/r/clusterhq/flocker-dataset-agent/>`_
+* `Container Agent <https://hub.docker.com/r/clusterhq/flocker-container-agent/>`_
+* `Control Service <https://hub.docker.com/r/clusterhq/flocker-control-service/>`_
+* `Docker Plugin <https://hub.docker.com/r/clusterhq/flocker-docker-plugin/>`_
+
+Before you install Flocker in containers, all authentication must be completed.
+This means that you will have generated the cluster certificate and key, the  node certificates and keys, and the API certificates and keys.
+
+In the example commands below, certificates are assumed to live in :file:`/etc/flocker` but other environments may choose to create containers that hold certificates and configuration, and share them amongst the Flocker service containers.
+The latter is not documented.
+
+For more information on generating certificates, please see  :ref:`authentication`. 
+
+Before you begin, you will need to make sure port ``4523`` is available for the control API, and port ``4524`` is available for agent nodes.
+
+Use the following steps to install Flocker using Docker containers:
+
+#. Run the Control Service:
+
+   * Run the following command on one of the hosts to create a volume for the control service configuration state to live in so it remains past the container lifecycle:
+
+     .. prompt:: bash $
+
+        docker run --name=flocker-control-volume -v /var/lib/flocker clusterhq/flocker-control-service true
+	 
+   * Run the following command on the same host to start the control-service:
+
+     .. prompt:: bash $
+
+        docker run --restart=always -d --net=host -v /etc/flocker:/etc/flocker --volumes-from=flocker-control-volume --name=flocker-control-service clusterhq/flocker-control-service
+
+#. Run the Container agent:
```

Comment:
hmmm, personally i havent seen/heard that. Seeing as where running the `container-agent` process it makes sense to call it so in the docs. Open to changing if I can find out own docs use this terminology, looking now.